specs/WPF_WebView2CompositionControl.md

```diff
@@ -0,0 +1,74 @@
+it in general in WPF here:
+https://dwayneneed.github.io/wpf/2013/02/26/mitigating-airspace-issues-in-wpf-applications.html
+
+Solving this issue for WebView2 necessitates moving away from the HwndHost and the windowed hosting model
+of WebView2, and instead use visuals to host the WebView2. To avoid compatibility issues and introducing
+new regressions in the existing control, we decided to create a new WebView2 WPF control, called the WebView2CompositionControl.
+
+# Examples
+In this example we create a WebView2CompositionControl in exactly the same way we would a regular
+WebView2 control in WPF. Additionally, we place a pink Label that appears ABOVE the WebView2CompositionControl.
+If you were to instead do this with the regular WebView2 control, the Label would be hidden behind it.
+```xml
+    <DockPanel>
+
+        <DockPanel
+            DockPanel.Dock="Top">
+            <Button
+                x:Name="ButtonGo"
+                DockPanel.Dock="Right"
+                Click="ButtonGo_Click"
+                Content="Go" />
+            <TextBox Name="addressBar"/>
+        </DockPanel>
+        <Grid>
+            <wv2:WebView2CompositionControl
+                Name="webView"
+                Source="https://www.microsoft.com"
+                AllowExternalDrop="False"/>
+            <Label
+                Content="This is a label over the WebView2!"
+                Background="LightPink"
+                Height="100"
+                Width="300"/>
+        </Grid>
+    </DockPanel>
+```
+![image](https://github.com/user-attachments/assets/023ad943-dca5-4f5e-a3f4-52258b35a6cd)
+
+
+# API Details
+If you are building a Windows Presentation Foundation (WPF) app and using the WebView2 control, you
+may find that your app runs into "airspace" issues, where the WebView2 control will always show up
+on top and hide any WPF elements in the same location, even if you try to specify the WPF elements
+to be above the WebView2 control (using visual tree order or the z-index property, for example). To
+solve this issue you should use the WebView2CompositionControl. 
+
+The WebView2CompositionControl is a drop-in replacement
+for the standard WPF WebView2 control - both the WebView2 control and the WebView2CompositionControl
+implement the [Microsoft.Web.WebView2.Wpf.IWebView2](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.wpf.iwebview2?view=webview2-dotnet-1.0.2739.15)
+interface, as well as derive from FrameworkElement (`FrameworkElement -> HwndHost -> WebView2`, and
+`FrameworkElement -> Control -> WebView2CompositionControl`). So anywhere you use
+WebView2, you can instead change to a WebView2CompositionControl without changing your
+code.
+
+We chose the name "WebView2CompositionControl" because it matches our naming convention
+for visual hosting WebView2, which uses the CoreWebView2CompositionController instead
+of the CoreWebView2Controller.
```

Comment:
Please add the C# definition of the class. It would also be helpful to link to the existing WebView2 control for comparison.

An easy way to get the C# def is with ILSpy to open up your DLL and copy it out

![image](https://github.com/user-attachments/assets/63b3c4c5-20ac-4647-a581-a32d003c0fe4)

Reply:
Unfortunately trying to copy from ILSpy includes all of the decompiled code as well, not just the definition. I haven't found a way to copy from the side navigation pane. The "public" filtering also only seems to affect the pane. Would screenshots of the pane help?

Reply:
The goal of this document and process is to review the API and its documentation so we need to include that in this document. A screenshot doesn't work because reviewers can't add comments to individual lines. I was imagining you could remove the decompiled code parts. I'll see if I can find an easier way to get that info

Reply:
Ok I'll try to get that. Normally that seems like a fine approach, but in this case it's a huge file :P

Reply:
Added. Also added some notes explicitly calling out the API differences between the two controls.